### DISCUSSION TITLE
Multiple subnets in Storage virtual Network Rules

### DISCUSSION THREAD

**Discussion by Git-Kamil** [Azure/ResourceModules#1710] [Q&A]
Hello,
I wonder if someone could help me. 
I'm using Micorosft.Storage template and I would like to add more than one subnet to  virtualNetworkRules. I tired to add them as the below code  but it don't work . Is it possible to add more than one subnet in the current template ?
Any help would be much appreciated.  

```json
"networkAcls": {
    "value": {
        "bypass": "AzureServices",
        "defaultAction": "Deny",
        "virtualNetworkRules": [
            {
                "subnet": "subnet01"
            }
            {
                "subnet": "subnet02"
            }
        ],
        "ipRules": []
    }
}
```

**Comment by eriqua** [comment]
Hello @Git-Kamil,

Yes, it is possible to specify multiple subnets for the storage network ACLs. You can refer to the storage account module readme section [networkAcls parameter usage](https://github.com/Azure/ResourceModules/blob/main/modules/Microsoft.Storage/storageAccounts/readme.md#parameter-usage-networkacls). 
This is the snippet I used to test:

```json
 "networkAcls": {
            "value": {
                "bypass": "AzureServices",
                "defaultAction": "Deny",
                "virtualNetworkRules": [
                    {
                        "id": "/subscriptions/<<subscriptionId>>/resourceGroups/validation-rg/providers/Microsoft.Network/virtualNetworks/adp-<<namePrefix>>-az-vnet-x-001/subnets/<<namePrefix>>-az-subnet-x-001",
                        "action": "Allow"
                    },
                    {
                        "id": "/subscriptions/<<subscriptionId>>/resourceGroups/validation-rg/providers/Microsoft.Network/virtualNetworks/adp-<<namePrefix>>-az-vnet-x-001/subnets/<<namePrefix>>-az-subnet-x-002",
                        "action": "Allow"
                    }
                ],
                "ipRules": [
                    {
                        "action": "Allow",
                        "value": "1.1.1.1"
                    }
                ]
            }
        }
```

Make sure that both subnets have a service endpoint configured for Microsoft.Storage.

Cheers

**Comment by AlexanderSehr** [comment]
@Git-Kamil , not that in JSON you need a comma `,` in between items of an array :)
